src/Collection/Functions/Result/DbalExpressionResult.php

```diff
@@ -161,11 +166,22 @@ public function withHavingArgs(string $havingExpression, array $havingArgs): Dba
 	}
 
 
+	public function collect(ExpressionContext $context): DbalExpressionResult
```

Comment:
Design decision - make the DbalExpressionResult configurable with collectFun or override it in custom impl for and/or?